conduit/plugins/importers/algod/algod_importer.go

```diff
@@ -228,12 +228,55 @@ func checkRounds(logger *logrus.Logger, catchpointRound, nodeRound, targetRound
 	return false, nil
 }
 
-func (algodImp *algodImporter) catchupNode(catchpoint string, targetRound uint64) error {
+func (algodImp *algodImporter) needsCatchup(targetRound uint64) bool {
+	if algodImp.mode == followerMode {
+		// If we are in follower mode, check if the round delta is available.
+		_, err := algodImp.getDelta(targetRound)
+		if err != nil {
+			algodImp.logger.Infof("Unable to fetch state delta for round %d: %s", targetRound, err)
+		}
+		return err != nil
+	} else {
+		// Otherwise just check if the block is available.
+		_, err := algodImp.aclient.Block(targetRound).Do(algodImp.ctx)
+		if err != nil {
+			algodImp.logger.Infof("Unable to fetch block for round %d: %s", targetRound, err)
+		}
+		// If the block is not available, we must catchup.
+		return err != nil
+	}
+}
+
+// catchupNode facilitates catching up via fast catchup, or waiting for the
+// node to slow catchup.
+func (algodImp *algodImporter) catchupNode(network string, targetRound uint64) error {
+	if !algodImp.needsCatchup(targetRound) {
+		return nil
+	}
```

Comment:
I recommend a log here as well. It may be a surprising (though a _pleasant_ surprise) for someone running the importer that the import starts happening immediately, even though they've configured for catching up.
It's also reassuring that they don't need to remove the catchup portion from the configuration if they stop and restart.